src/herder/HerderImpl.cpp

```diff
@@ -416,6 +410,12 @@ HerderImpl::recvSCPEnvelope(SCPEnvelope const& envelope)
         return Herder::ENVELOPE_STATUS_DISCARDED;
     }
 
+    if (envelope.statement.nodeID == getSCP().getLocalNode()->getNodeID())
+    {
+        CLOG(DEBUG, "Herder") << "recvSCPEnvelope: skipping own message";
+        return Herder::ENVELOPE_STATUS_SKIPPED_SELF;
```

Comment:
Why did this condition _move_ from higher up to down here, in addition to changing return code?

Reply:
Like I mentioned in a previous comment (and in the commit message), this allows to never flood SCP messages that are not in the right range - without this change, we can't tell if the message is old or not, which is the only criteria we care about for flooding.

Reply:
Ok, I think I understand now: previously both own-message and out-of-range cases resulted in early-return and continue-to-flood. The change here is to _keep_ own-message in that case (early-return and continue-to-flood) but to sharpen the out-of-range case to result in early-return and inhibit-flooding, which means that range test has to happen earlier as well as return a different code than the own-message case.

Phew! Thanks for the explanation. Approved!